shark_turbine/kernel/wave/hoisting.py

```diff
@@ -0,0 +1,31 @@
+from ...support.logging import get_logger
+from shark_turbine.kernel._support.tracing import CapturedTrace
+import torch.fx as fx
+from ..ops.wave_ops import *
+from .address_spaces import *
+
+logger = get_logger("turbine.wave.hoisting")
+
+
+def get_allocs_(graph: fx.Graph) -> list[CustomOp]:
+    return [
+        custom_node
+        for node in graph.nodes
+        if isinstance((custom_node := get_custom(node)), Allocate)
+    ]
```

Comment:
nit: If we provide a return type annotation it should be as strict as possible, i.e: `list[Allocate]`

Why does the name have a trailing `_`?

Reply:
Good point, will update. I was trying to follow your convention which I thought was to mark functions not to be exposed with a trailing underscore, but maybe that wasn't the intention. What was your convention? I can remove the underscore here.

Reply:
Ah, I see! I use trailing `_` only when I need it to avoid a name clash with an existing Python keyword.
The Python convention is that private functions should be prefixed with `_`.